graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```
@@ -147,55 +147,64 @@ class InputTypeGenerator(config: CodeGenConfig, document: Document) : BaseDataTy
+                is ObjectValue -> {
+                    val inputObjectDefinition = inputTypeDefinitions.first {
+                        val expectedCanonicalClassName = config.typeMapping[it.name] ?: "${config.packageNameTypes}.${it.name}"
+                        expectedCanonicalClassName == type.className.canonicalName()
+                    }
+                    if (value.objectFields.isEmpty()) {
+                        return CodeBlock.of("new $type()")
+                    } else {
+                        CodeBlock.of(
+                            "new $type(){{" + value.objectFields.joinToString("") { objectProperty ->
+                                val argumentType =
+                                    checkNotNull(inputObjectDefinition.inputValueDefinitions.find { it.name == objectProperty.name }) {
+                                        "Property \"${objectProperty.name}\" does not exist in input type \"${inputObjectDefinition.name}\""
+                                    }
+                                val argumentValue = generateCode(
+                                    objectProperty.value,
+                                    typeUtils.findReturnType(argumentType.type),
+                                    inputTypeDefinitions
+                                )
+                                "set${objectProperty.name.replaceFirstChar { it.uppercaseChar() }}($argumentValue);"
+                            } + "}}"
+                        )
+                    }
                 }
+                else -> CodeBlock.of("\$L", value)
             }
-            else -> CodeBlock.of("\$L", value)
-        }
     }
 
-    private fun checkAndGetLocaleValue(value: StringValue, type: JavaTypeName): String? {
-        if (type.toString() == "java.util.Locale") return "Locale.forLanguageTag(\"${value.value}\")"
-        return null
+    private fun checkAndGetLocaleCodeBlock(value: Value<out Value<*>>, type: JavaTypeName): CodeBlock? {
```

This change seems mostly unrelated. I assume it prevents a crash in the event a non-string is passed in? Would be good to add a test case for.

---

Also, assuming it does prevent a crash, perhaps rather than returning `null` it'd be better to throw some sort of exception with a friendly error message. If someone has set `someLocale: Locale = 123` in their schema, it's probably best we let them know rather than just silently not do anything.

---

I think you're correct. Will do the changes.